src/read_utils.jl

```diff
@@ -2,6 +2,45 @@ using DelimitedFiles
 
 # ---------------------------------------------------------------------------------------- #
 
+struct BandSummary
+    topology :: TopologyKind
+    n :: Vector{<:Integer}
+    irlabs :: Vector{<:Integer}
```

Comment:
Earlier comment also applies here (`Vector{<:Integer}` is not a concrete type):
```suggestion
    n :: Vector{Int}
    irlabs :: Vector{Int}
```